src/Explorer/Controls/Settings/SettingsComponent.tsx

```diff
@@ -0,0 +1,2588 @@
+                this.state.rupm.current === Constants.RUPMStates.off ? "selectedRadio" : "unselectedRadio"
+              }`}
+            >
+              Off
+            </label>
+            <input
+              type="radio"
+              name="rupm"
+              value={Constants.RUPMStates.off}
+              className="radio"
+              id={this.rupmOffId}
+              onChange={this.onRupmChange}
+              checked={this.state.rupm.current === Constants.RUPMStates.off}
+            />
+          </div>
+        </div>
+      </>
+    );
+  };
+
+  private getScaleComponent = (): JSX.Element => {
+    return (
+      <>
+        <AccessibleElement
+          as="div"
+          className="scaleDivison"
+          onClick={this.toggleScale}
+          onActivated={this.toggleScale}
+          aria-expanded={this.state.scaleExpanded}
+          role="button"
+          tabIndex={0}
+          aria-label="Scale"
+          aria-controls="scaleRegion"
+        >
+          {!this.state.scaleExpanded && (
+            <span className="themed-images" id="ExpandChevronRightScale">
+              <img
+                className="imgiconwidth ssExpandCollapseIcon ssCollapseIcon "
+                src={TriangleRight}
+                alt="Show scale properties"
+              />
+            </span>
+          )}
+
+          {this.state.scaleExpanded && (
+            <span className="themed-images" id="ExpandChevronDownScale">
+              <img className="imgiconwidth ssExpandCollapseIcon " src={TriangleDown} alt="Hide scale properties" />
+            </span>
+          )}
+
+          <span className="scaleSettingTitle">Scale</span>
+        </AccessibleElement>
+
+        {this.state.scaleExpanded && (
+          <div className="ssTextAllignment" id="scaleRegion">
+            {!this.isAutoScaleEnabled() && (
+              <>
+                {this.getThroughputInputComponent()}
+                <div className="storageCapacityTitle throughputStorageValue">{this.getStorageCapacityTitle()}</div>
+              </>
+            )}
+
+            {this.isRupmVisible() && this.getRupmComponent()}
+
+            {/*<!-- TODO: Replace link with call to the Azure Support blade -->*/}
+            {this.isAutoScaleEnabled() && (
+              <div>
+                <div className="autoScaleThroughputTitle">Throughput (RU/s)</div>
+                <input className="formReadOnly collid-white" readOnly aria-label="Throughput input" />
+                <div className="autoScaleDescription">
+                  Your account has custom settings that prevents setting throughput at the container level. Please work
+                  with your Cosmos DB engineering team point of contact to make changes.
+                </div>
+              </div>
+            )}
+          </div>
+        )}
+      </>
+    );
+  };
+
+  private getConflictResolutionModeComponent = (): JSX.Element => {
```

Comment:
Now that we have these inputs nicely isolated to separate functions. Can we create separate `ReactComponents` for each of them?
If they are "dumb" components (ie without elaborate biz logic, etc) you can make them a react function component (see `FeaturePanelComponent` for example). A react function component is essentially one single big function that renders some props. You can also add state to it and side effects. Avoid nested functions (that are harder to unit test).

Reply:
Once separate in components, how hard would it be to convert to fluent-ui? Since we're in this code, we might as well go all the way in fixing/refactor.